### PR TITLE
Api testing

### PR DESCRIPTION
**Pull request by Eternal-Flame085**
- Created spec for top movies
- Created route
- Created Movies controller and MoviesApiService controller
- Created view for top movies
- Needs more testing

### CODE REVIEW

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -1,23 +1,9 @@
...
-      response = conn.get("/3/search/movie?&page=#{page_number}") do |req|
-        req.params[:query] = query
-      end
-
-      title_data = JSON.parse(response.body, symbolize_names: true)
-
-      @titles << title_data[:results]
-    end
+    @titles = MovieApiService.movie_search(params[:movie_title])
+  end
```
Comment (Stearnzy) [inline app/controllers/movies_controller.rb:4]: Thanks for moving these

File: spec/features/movie/movie_spec.rb.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
```
Comment (Stearnzy) [inline spec/features/movie/movie_spec.rb.rb]: Perhaps we can rename this spec file

File: app/movie/movie_api_service.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    titles = []
+    page_number = 0
+
+    2.times do
+      page_number += 1
+      response = conn.get("/3/search/movie?&page=#{page_number}") do |req|
+        req.params[:query] = query
+      end
+
+      title_data = JSON.parse(response.body, symbolize_names: true)
```
Comment (stillsheryl) [inline app/movie/movie_api_service.rb]: Can we also use the parse_data method here?

**Review by Stearnzy — APPROVED** [review]

File: spec/models/movie_api_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
```
Comment (stillsheryl) [inline spec/models/movie_api_spec.rb:2]: I'm not sure what the convention is, but have seen something like this before:

```
describe MovieApiService do
  context "instance methods" do
    describe ".conn" do
      it "returns 200 status" do
        api_response = MovieApiService.conn.get("/3/movie/top_rated")
        expect(api_response.status).to eq(200)
      end
   end
end
```

**Review by stillsheryl — APPROVED** [review]
Looks good!

**Pull request merged**